tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -38,185 +43,235 @@
+    Args:
+        host_manager (HostManager): HostManager fixture.
     """
-    configurations = {}
-
-    for host in host_manager.get_group_hosts('all'):
-        if host in host_manager.get_group_hosts('agent'):
-            configurations[host] = load_configuration_template(configurations_paths['agent'], [{}], [{}])
-        elif host in host_manager.get_group_hosts('manager'):
-            configuration_template = load_configuration_template(configurations_paths['manager'], [{}], [{}])
-
-            # Replace placeholders by real values
-            manager_index = host_manager.get_group_hosts('manager').index(host) + 2
-            indexer_server = host_manager.get_group_hosts('indexer')[0]
-            indexer_server_variables = host_manager.get_host_variables(indexer_server)
-            configuration_variables = {
-                'INDEXER_SERVER': indexer_server_variables['ip'],
-                'FILEBEAT_ROOT_CA': '/etc/pki/filebeat/root-ca.pem',
-                'FILEBEAT_CERTIFICATE': f"/etc/pki/filebeat/node-{manager_index}.pem",
-                'FILEBEAT_KEY': f"/etc/pki/filebeat/node-{manager_index}-key.pem"
-            }
-            configuration_template_str = str(configuration_template)
-
-            for key, value in configuration_variables.items():
-                configuration_template_str = configuration_template_str.replace(key, value)
-
-            configurations[host] = ast.literal_eval(configuration_template_str)
-
-    return configurations
-
-@pytest.fixture(scope='module', autouse=True)
-def install_npm(host_manager: HostManager):
-    """Check and install npm if not already installed"""
-
-    node_version = "v21.7.1"
-    node_package_url = f"https://nodejs.org/dist/{node_version}/node-{node_version}.pkg"
-
-    target_os_groups = ['macos']
-
-    for group in target_os_groups:
-        for host in host_manager.get_group_hosts(group):
-            # Check if Node and npm is installed
-            logger.info(f"Checking and installing npm on {host}")
-            node_check_command = "PATH=/usr/local/bin:$PATH && command -v node"
-            node_check_result = host_manager.get_host(host).ansible(
-                "shell",
-                node_check_command,
-                become=True,
-                become_user='vagrant',
-                check=False
-            )
-            logger.info(f"Node check result on {host}: {node_check_result}")
-            # Install node if it is not already installed.
-            if node_check_result['rc'] != 0:
-                logger.info(f"Installing Node.js and npm using package: {node_package_url}")
-
-                # Use the install_package method to handle the installation.
-                install_result = host_manager.install_package(host, node_package_url, system='macos')
+    logging.error(
+        "Save the Wazuh indexer username and password into the Wazuh manager keystore"
+    )
+    save_indexer_credentials_into_keystore(host_manager)
```

Comment:
This could be in the conftest

Reply:
Done c55a05626962807460140be1b3fddf961fb87fad